tests/test_unittest/test_cli.py

```diff
@@ -172,47 +137,59 @@ def test_backend_choice(backend_name):
     assert backend_name in cli.BackendChoice.get_backend_names()
 
 
-def test_unenabled_backend_choice(clirunner, nocli_backend, simple_stencil):
+def test_gen_unenabled_backend_choice(clirunner, nocli_backend, simple_stencil, tmp_path):
     """
     Test the --backend option when an unenabled backend name is passed.
-
-    The :py:func:`nocli_backend` fixture temporarily injects a dummy backend named "nocli", which
-    is not CLI enabled (and provides no functionality in fact).
-
-    Actions:
-
-        .. code-block:: bash
-
-            $ gtpyc --backend=nocli <simple_stencil_path>
-
-    Outcome:
-
-        An error message warns the user that the chosen backend is not enabled for CLI.
-        The command aborts with error code 2.
-
     """
-    result = clirunner.invoke(cli.gtpyc, ["--backend=nocli", str(simple_stencil.absolute())])
+    result = clirunner.invoke(
+        cli.gtpyc,
+        [
+            "gen",
+            "--backend=nocli",
+            str(simple_stencil.absolute()),
+            "--output-path",
+            str(tmp_path / "test_gen_unenabled"),
+        ],
+    )
 
     assert result.exit_code == 2
     assert re.findall(r".*Backend is not CLI-enabled\..*", result.output)
 
 
-def test_enabled_backend_choice(clirunner, simple_stencil, backend_name, backend_enabled):
+def test_gen_enabled_backend_choice(
+    clirunner, simple_stencil, backend_name, backend_enabled, tmp_path
+):
     """
     Test an enabled backend.
+    """
+    result = clirunner.invoke(
+        cli.gtpyc,
+        [
+            "gen",
+            "--backend",
+            backend_name,
+            str(simple_stencil.absolute()),
+            "--output-path",
+            str(tmp_path / "test_gen_unenabled"),
+        ],
+        catch_exceptions=False,
+    )
 
-    Actions:
-
-        .. code-block:: bash
-
-            $ gtpyc --backend <parametrized backend_name> <simple_stencil path>
+    if backend_enabled:
+        assert result.exit_code == 0
+    else:
+        assert result.exit_code == 2
 
-    Outcome:
 
-        The command writes the computation source to the current directory.
-    """
+def test_gen_gtx86(clirunner, simple_stencil, tmp_path):
+    """Only generate the c++ files."""
+    output_path = tmp_path / "test_gen_gtx86"
     result = clirunner.invoke(
-        cli.gtpyc, ["--backend", backend_name, str(simple_stencil.absolute())]
+        cli.gtpyc,
+        ["gen", f"--output-path={output_path}", "--backend=gtx86", str(simple_stencil)],
+        catch_exceptions=False,
     )
-
-    assert result.exit_code == 0 if backend_enabled else 2
+    assert result.exit_code == 0
+    src_path = output_path / "init_1_src"
+    src_files = [path.name for path in src_path.iterdir()]
+    assert set(["computation.hpp", "computation.cpp"]) == set(src_files), result.output
```

Comment:
Contrary to before fine with me here since this is a specific test for the gtx86 backend.

Reply:
Thing is all the gt* backends have the names hardcoded in the same place.